services-custom/s3-transfermanager/src/main/java/software/amazon/awssdk/custom/s3/transfer/internal/S3CrtUtils.java

```diff
@@ -238,19 +237,28 @@ private static String encodedQueryString(Map<String, List<String>> rawQueryParam
                            .orElse("");
     }
 
-    private static void addRequestOverrideConfiguration(AwsRequestOverrideConfiguration requestOverrideConfiguration,
-                                                        Consumer<HttpHeader[]> headersConsumer,
-                                                        Consumer<String> queryParametersConsumer) {
-        throwExceptionForUnsupportedConfigurations(requestOverrideConfiguration);
+    private static void processRequestOverrideConfiguration(Optional<AwsRequestOverrideConfiguration> optionalConfig,
+                                                            Consumer<String> queryParametersConsumer) {
+        if (optionalConfig.isPresent()) {
+            AwsRequestOverrideConfiguration requestOverrideConfiguration = optionalConfig.get();
 
-        if (!requestOverrideConfiguration.headers().isEmpty()) {
-            HttpHeader[] httpHeaders = createHttpHeaders(requestOverrideConfiguration.headers());
-            headersConsumer.accept(httpHeaders);
+            throwExceptionForUnsupportedConfigurations(requestOverrideConfiguration);
+            if (!requestOverrideConfiguration.rawQueryParameters().isEmpty()) {
+                String encodedQueryString = encodedQueryString(requestOverrideConfiguration.rawQueryParameters());
+                queryParametersConsumer.accept(encodedQueryString);
+            }
         }
+    }
 
-        if (!requestOverrideConfiguration.rawQueryParameters().isEmpty()) {
-            String encodedQueryString = encodedQueryString(requestOverrideConfiguration.rawQueryParameters());
-            queryParametersConsumer.accept(encodedQueryString);
+    private static void addCustomHeaders(Optional<AwsRequestOverrideConfiguration> requestOverrideConfiguration,
+                                         Consumer<HttpHeader[]> headersConsumer) {
+        List<HttpHeader> crtHeaders = new ArrayList<>();
+        crtHeaders.add(new HttpHeader(HEADER_USER_AGENT, USER_AGENT_STRING));
```

Comment:
Won't this override the other user-agent headers?

Reply:
This is no existing `User-Agent` header set on the CRT side, I think. I'll double check with CRT team. S3 CRT client doesn't have APIs to append user agent string unfortunately.

Reply:
Yeah, confirmed that the CRT S3 client will add a User-Agent header if one doesn't exist or append to the existing User-Agent header

Reply:
If the CRT doesn't set the user agent, we should set it. We don't want to lose all of the other information we usually include in the user agent...

Reply:
Yeah, good point, will add other stuff as well